plugin/plugin.go

```diff
@@ -1,186 +0,0 @@
-package plugin
```

Comment:
What drove the decision to move this back into the operator package?

Reply:
That decision is probably no longer valid. Initially, it was because I was planning on refactoring it so that a plugin was just treated as an operator, but that ended up being way more complex than I expected. I'll move it back out.